### DISCUSSION TITLE
How to use telescope extensions?

### DISCUSSION THREAD

**Discussion by egavriloff** [NvChad/NvChad#1208] [Q&A] [answered]
I want to use plugin [session-lens](https://github.com/rmagatti/session-lens). 

When I start vim, if I run the `:SearchSession` command, I get an error:
```
E5108: Error executing lua ...m/site/pack/packer/start/packer.nvim/lua/packer/load.lua:165: Vim(echomsg):E114: Missing quote: "Error in packer_compiled: ...m/site/pac
k/packer/start/packer.nvim/lua/packer/load.lua:139: Vim(let):E5108: Error executing lua ...escope/_extensions/session-lens/session-lens-actions.lua:2: module 'telesco
pe.actions.state' not found:
stack traceback:
        [C]: in function 'cmd'
        ...m/site/pack/packer/start/packer.nvim/lua/packer/load.lua:165: in function <...m/site/pack/packer/start/packer.nvim/lua/packer/load.lua:161>
        [string ":lua"]:1: in main chunk
```

But if you first execute `:Telescope find_files` and then `:SearchSession` - it works...

![ezgif com-gif-maker](https://user-images.githubusercontent.com/18679073/173640452-b17bd327-920a-4d82-914b-e3f809ae2926.gif)

Config:
<img width="633" alt="image" src="https://user-images.githubusercontent.com/18679073/173640668-97e983c9-9f1f-4706-b69a-ef8d650db251.png">

can I force load Telescope?

**Comment by siduck** [comment, marked as answer]
@eugene-gavriloff make sure u override telescope config  and include your extension in this list here first
![image](https://user-images.githubusercontent.com/59060246/173712760-2435c6e7-e26f-4fd3-b0f2-56569ed70b50.png)

**Reply by egavriloff** [reply]
thanks! it works !!!

**Reply by chikko80** [reply]
@siduck 

This doesn't work for me.. Totally new to NvChad trying to install a telescope extension.
I put this:

```lua
M.telescope = {
  extensions_list = { "themes", "terms", "fzy-native" , "zoxide"},
}
```

in custom/configs/overrides.lua, and did a lazy = false load in the custom/plugins.lua.
Plugins get loaded on NvChadUpdate but for example ":Telescope zoxide" is not found

Any idea? probably just a noob question

**Reply by siduck** [reply]
@chikko80 please read our plugin docs properly https://nvchad.com/docs/config/plugins

```lua
{
    "nvim-telescope/telescope.nvim",
    dependencies = "jvgrootveld/telescope-zoxide",

    opts = {
      extensions_list = { "themes", "terms", "zoxide" },
      extensions = {
        zoxide = {
          prompt_title = "[ Walking on the shoulders of TJ ]",
        },
      },
    },
  },
  ```

**Reply by chikko80** [reply]
Thanks i missed the lazy.vim docs. Thats why i was confused since there isn't much docs under your link.

**Reply by Endermanbugzjfc** [reply]
> ```lua
> {
>     "nvim-telescope/telescope.nvim",
>     dependencies = "jvgrootveld/telescope-zoxide",
> 
>     opts = {
>       extensions_list = { "themes", "terms", "zoxide" },
>       extensions = {
>         zoxide = {
>           prompt_title = "[ Walking on the shoulders of TJ ]",
>         },
>       },
>     },
>   },
> ```

That saved me... (I don't think this is described in the updated doc.)

**Comment by danawaz** [comment]
I can't seem to get toggletasks to load using the answer above. Here is my config:

```
{
    "nvim-telescope/telescope.nvim",
    dependencies = "jedrzejboczar/toggletasks.nvim",

    opts = {
      extensions_list = { "themes", "terms", "toggletasks" },
      extensions = {
        toggletasks = {
          prompt_title = "[ toggletasks ]",
        },
      },
    },
  },

  {
    "akinsho/toggleterm.nvim",
    config = function()
      require("toggleterm").setup()
    end,
  },

  {
    "jedrzejboczar/toggletasks.nvim",
    config = function()
      require("toggletasks").setup()
    end,
  },
```

When I call `Telescope toggletasks` I get the following error:
<img width="1440" alt="Screenshot 2023-04-18 at 11 22 57 AM" src="https://user-images.githubusercontent.com/4193942/232825210-a861d2d1-854e-40bc-bafd-074c4d961dcd.png">

I can see that toggletasks is installed in lazy:
<img width="1440" alt="Screenshot 2023-04-18 at 11 24 30 AM" src="https://user-images.githubusercontent.com/4193942/232825708-55781471-3a1a-4738-8cdf-0763cba56d48.png">

I have a separate vim config without Nvchad and there toggletasks loads fine without any issues. Kindly guide what mistake I am making here.

**Reply by siduck** [reply]
@eugene-gavriloff  load toggletasks as dependencies

```lua
{
    "nvim-telescope/telescope.nvim",
    dependencies = {
    "jedrzejboczar/toggletasks.nvim",
    config = function()
      require("toggletasks").setup()
    end,
  },

    opts = {
      extensions_list = { "themes", "terms", "toggletasks" },
      extensions = {
        toggletasks = {
          prompt_title = "[ toggletasks ]",
        },
      },
    },
  },

  {
    "akinsho/toggleterm.nvim",
    config = function()
      require("toggleterm").setup()
    end,
  },
  ```

**Reply by danawaz** [reply]
@siduck thanks for the prompt response, but that does not seem to solve the problem, I still get the same error.
<img width="1440" alt="Screenshot 2023-04-18 at 2 43 22 PM" src="https://user-images.githubusercontent.com/4193942/232873796-58bd9972-3219-4a6b-a09c-075c20ceb5e1.png">

Toggletasks still shows up in Telescope:
 
<img width="1440" alt="Screenshot 2023-04-18 at 2 43 46 PM" src="https://user-images.githubusercontent.com/4193942/232873763-15f90377-0f1b-43f9-a599-a222e1cec8f2.png">

**Reply by danawaz** [reply]
@siduck Do you have any leads to why this is not working? I tried using `nvim-telescope/telescope-dap.nvim` for dap settings in telescope and that does not seem to work either. This plugin also require the use of `load_extension` to be used with telescope.

**Reply by siduck** [reply]
@dani199221 i'll test the plugin here today, sorry for the delay!

**Reply by danawaz** [reply]
@siduck  Were you able to find why the load_extension does not work?

**Reply by siduck** [reply]
@dani199221  I checked its readme today and I see the "rocks" opt,  and fyi lazy.nvim doesnt support luarocks https://github.com/folke/lazy.nvim/issues/37 . 

Also toggletasks's creator himself suggests using overseer.nvim https://github.com/jedrzejboczar/toggletasks.nvim/issues/16#issuecomment-1460742344.

For tasks i personally use our own plugin "nvterm" , check its docs https://github.com/NvChad/nvterm#additional-functionality
